src/rez/plugin_managers.py

```diff
@@ -235,7 +239,7 @@ def config_schema(self):
                 deep_update(d, d_)
         return dict_to_schema(d, required=True, modifier=expand_system_vars)
 
-    def create_instance(self, plugin, **instance_kwargs):
+    def create_instance(self, plugin, **instance_kwargs) -> "RezPluginType":
```

Comment:
This doesn't return a `RezPluginType` instance. It will return something like `rezplugins.build_system.cmake.CMakeBuildSystem`. I don't think we have an easy way today to get all the possible base classes for each plugin type. It should be possible, but it'll require changes to the plugin system, or some more thoughts to maybe use generics or whatever type magic we'll need to get this to work.

Reply:
@JeanChristopheMorinPerso Ah right good point. Would it be alright to type hint these just as `object` instead for now? That at least gives us some kind of information.